sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -771,44 +781,65 @@ export class MessageReceiver extends LinkEntity {
           this.name = options.name;
         }
 
-        this.isConnecting = true;
-        await this._negotiateClaim();
-        if (!options) {
-          options = this._createReceiverOptions();
-        }
-        log.error(
-          "[%s] Trying to create receiver '%s' with options %O",
-          connectionId,
-          this.name,
-          options
-        );
+        await defaultLock.acquire(this._openLock, async () => {
+          if (this.wasCloseInitiated) {
+            throw createNonRetryableError(
+              getReceiverClosedErrorMsg(
+                this._context.entityPath,
+                this._context.clientType,
+                this._context.isClosed,
+                undefined
+              )
+            );
+          }
 
-        this._receiver = await this._context.namespace.connection.createReceiver(options);
-        this.isConnecting = false;
-        log.error(
-          "[%s] Receiver '%s' with address '%s' has established itself.",
-          connectionId,
-          this.name,
-          this.address
-        );
-        log[this.receiverType](
-          "Promise to create the receiver resolved. " + "Created receiver with name: ",
-          this.name
-        );
-        log[this.receiverType](
-          "[%s] Receiver '%s' created with receiver options: %O",
-          connectionId,
-          this.name,
-          options
-        );
-        // It is possible for someone to close the receiver and then start it again.
-        // Thus make sure that the receiver is present in the client cache.
-        if (this.receiverType === ReceiverType.streaming && !this._context.streamingReceiver) {
-          this._context.streamingReceiver = this as any;
-        } else if (this.receiverType === ReceiverType.batching && !this._context.batchingReceiver) {
-          this._context.batchingReceiver = this as any;
-        }
-        await this._ensureTokenRenewal();
+          if (this.isOpen()) {
+            return;
+          }
+
+          this.isConnecting = true;
```

Comment:
The `isConnecting` property gets checked in multiple places to make the decision of whether to call `onDetached()` or not so that there are not multiple attempts being made at recovering the link. 

With the change in this PR, `isConnecting` is now being set to true only after the lock is acquired, resulting in potential multiple calls to `onDetached` now being a possibility.

Recently,

Reply:
Though recently in #8401, @chradek did add a new `_isDetaching` flag to ensure that `onDetached()` is a no-op if it gets called multiple times..

Reply:
But multiple calls to `onDetached()` will add to the noise in the logs

So, I would recommend moving the `this.isConnecting = true` to before the lock is acquired

Reply:
I'll take a deeper look at this. Your comments have me realize I need to consider the onDetached/open/close in tandem and I don't think I've done that enough.